tutorials/05_segmentation_camera.md

```diff
@@ -0,0 +1,460 @@
+
+    print("Press Any Key to Continue ... ESC to exit")
+    if cv2.waitKey(0) == 27:
+        break
+
+cv2.destroyAllWindows()
+```
+
+Make sure you have all of the dependencies that are needed in order to run this Python script:
+* https://pypi.org/project/opencv-python/
+* https://numpy.org/install/
+
+Run the script, setting the `--path` argument to what you specified for the `<save>` tag in the SDF file.
+Since we set the panoptic save path to `segmentation_data/instance_camera`, we'd run the following command to view the panoptic segmentation data:
+```
+python3 segmentation_visualizer.py --path segmentation_data/instance_camera
+```
+
+You will see 4 windows: `image`, `labels_map`, `colored_map`, and `colored_image` (which is a combination of the image and colored_map).
+
+@image html files/segmentation_camera/segmentation_images.png width=900px
+
+For panoptic/instance segmentation, to parse the `labels_map`, click on any pixel on the `labels_map` window to see the `label` and `instance count` of that pixel.
+
+
+## Processing the segmentation sensor via ign-transport
+It's possible to process the segmentation data in real time via `ign-transport`.
+You will need to which topics to subscribe to in order to receive this information.
+
+Consider the following SDF snippet from the segmentation camera:
+```xml
+    <topic>segmentation</topic>
+```
+
+In this scenario, the sensor data will publish the label map data to `segmentation/labels_map`, and the colored map data to `segmentation/colored_map`.
+We can write some c++ code that subscribes to these topics:
+
+```cpp
+#include <cstdint>
+
+#include <ignition/msgs.hh>
+#include <ignition/rendering.hh>
+#include <ignition/transport.hh>
+
+void OnNewLabelMap(const ignition::msgs::Image &_msg)
+{
+  auto width = _msg.width();
+  auto height = _msg.height();
+  auto buffer = _msg.data();
+
+  for (uint32_t i = 0; i < height; ++i)
+  {
+    for (uint32_t j = 0; j < width; ++j)
+    {
+      auto index = (i * width + j) * 3;
+
+      // label id of the pixel is in the last channel
+      auto label = buffer[index + 2];
+
+      // In case of semantic segmentation, instance count is ignored.
+      //
+      // For panoptic segmentation, we can get the instance count as follows:
+      // 16 bit value for instance count
+      auto instanceCount1 = buffer[index + 1];
+      auto instanceCount2 = buffer[index];
+      // get the 16-bit decimal value from the two 8-bit values by left-shifting
+      // instanceCount1 by 8 bits
+      auto instanceCount = instanceCount1 * 256 + instanceCount2;
+    }
+  }
+}
+
+void OnNewColoredMap(const ignition::msgs::Image &_msg)
+{
+  auto width = _msg.width();
+  auto height = _msg.height();
+  auto buffer = _msg.data();
+
+  for (uint32_t i = 0; i < height; ++i)
+  {
+    for (uint32_t j = 0; j < width; ++j)
+    {
+      auto index = (i * width + j) * 3;
+      auto r = buffer[index + 2];
+      auto g = buffer[index + 1];
+      auto b = buffer[index];
```

Comment:
Indexing has been updated to follow RGB instead of GBR in order to match https://github.com/ignitionrobotics/ign-rendering/pull/419: see https://github.com/ignitionrobotics/ign-sensors/pull/133/commits/8642b60fdac3e03b8712d6a39b62737c49bcc9bb